core/src/test/java/org/infinispan/distribution/groups/TransactionalGetGroupKeysTest.java

```diff
@@ -140,8 +140,7 @@ public void testGetGroupsWithConcurrentReplace() throws Exception {
 
       testCache.primaryOwner.put(key(1), value(-1));
 
-      if ((factory == TestCacheFactory.PRIMARY_OWNER || factory == TestCacheFactory.BACKUP_OWNER) &&
-            isolationLevel == IsolationLevel.READ_COMMITTED) {
+      if (isolationLevel == IsolationLevel.READ_COMMITTED) {
```

Comment:
I got lost in the conditionals :) The key *is* in context after we call `getGroup()` the first time. On the remote node, it always iterates whole DC and finds all keys that are in given group, and sends this result back. In RC the context is overwritten, in RR it is not.
It's possible that in the past, the context was not overwritten neither in RC and therefore the test had to contain that condition - I haven't checked that.